clusterloader/framework/util.go

```diff
@@ -79,3 +106,248 @@ func CreatePods(f *framework.Framework, appName string, ns string, labels map[st
 		}
 	}
 }
+
+// GetTuningSet matches the name of the tuning set defined in the project and returns a pointer to the set
+func GetTuningSet(tuningSets []TuningSetType, podTuning string) (tuning *TuningSetType) {
+	if podTuning != "" {
+		// Iterate through defined tuningSets
+		for _, ts := range tuningSets {
+			// If we have a matching tuningSet keep it
+			if ts.Name == podTuning {
+				tuning = &ts
+				return
+			}
+		}
+		framework.Failf("No pod tuning found for: %s", podTuning)
+	}
+	return nil
+}
+
+// Server is the webservice that will syncronize the start and stop of Pods
+func Server(c *PodCount) error {
+	http.HandleFunc("/start", handleStart(startHandler, c))
+	http.HandleFunc("/stop", handleStop(stopHandler, c))
+
+	server := &http.Server{Addr: ":8081", Handler: nil}
+
+	ln, err := net.Listen("tcp", server.Addr)
+	if err != nil {
+		return err
+	}
+
+	go server.Serve(ln)
+	fmt.Println("Listening on port", server.Addr)
+	select {
+	case <-c.Shutdown:
+		fmt.Println("Shutdown server")
+		ln.Close()
+		return err
+	}
+}
+
+func handleStart(fn http.HandlerFunc, c *PodCount) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		c.Started++
+		fn(w, r)
+		fmt.Printf("Started pods: %d, Stopped pods: %d\n", c.Started, c.Stopped)
+	}
+}
+
+func startHandler(w http.ResponseWriter, r *http.Request) {
+	fmt.Fprintln(w, "Hello")
+}
+
+func handleStop(fn http.HandlerFunc, c *PodCount) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		c.Stopped++
+		fn(w, r)
+		fmt.Printf("Started pods: %d, Stopped pods: %d\n", c.Started, c.Stopped)
+		if c.Stopped == c.Started && c.Stopped > 0 {
+			c.Shutdown <- true
+		}
+	}
+}
+
+func stopHandler(w http.ResponseWriter, r *http.Request) {
+	fmt.Fprintln(w, "Goodbye")
+}
+
+// firstLowercase conversts the first letter of a string to lowercase.
+func firstLowercase(s string) string {
+	a := []rune(s)
+	a[0] = unicode.ToLower(a[0])
+	return string(a)
+}
+
+// convertVariables takes our loaded struct and converts it into a map[string]string.
+func convertVariables(params ParameterConfigType) map[string]string {
+	m := structs.Map(params)
+	values := make(map[string]string)
+	for k, v := range m {
+		k = firstLowercase(k)
```

Comment:
Why this is needed? If names are case sensitive we should use this, if they're not just cast whole key.